arango/http.py

```diff
@@ -63,12 +70,77 @@ def send_request(
         raise NotImplementedError
 
 
-class DefaultHTTPClient(HTTPClient):
-    """Default HTTP client implementation."""
+class DefaultHTTPAdapter(HTTPAdapter):
+    """Default transport adapter implementation
+
+    :param pool_connections: The number of urllib3 connection pools to cache.
+    :type pool_connections: int
+    :param pool_maxsize: The maximum number of connections to save in the pool.
+    :type pool_maxsize: int
+    :param pool_timeout: Socket timeout in seconds for each individual connection.
+    :type pool_timeout: int | float | None
+    :param kwargs: Additional keyword arguments passed to the HTTPAdapter constructor.
+    :type kwargs: Any
+    """
+
+    def __init__(
+        self,
+        pool_connections: int = DEFAULT_POOLSIZE,
+        pool_maxsize: int = DEFAULT_POOLSIZE,
+        pool_timeout: Union[int, float, None] = DEFAULT_POOL_TIMEOUT,
```

Comment:
This is confusingly named. There is an argument to urlopen called `pool_timeout` which is the timeout to wait for a connection from the pool and only matters if `blocking=True`. This has nothing to do with the actual connections, just whether requests block to get a cached connection or just make a new one if all cached connections are busy.

You're mapping this to `timeout` which is a ConnectionPool argument that sets the default connect/read timeout for each connection. But because you set `timeout` on each request, this pool level default will always be ignored